packages/chat-stateful-client/review/chat-stateful-client.api.md

```diff
@@ -21,14 +22,38 @@ export type ChatClientState = {
     threads: {
         [key: string]: ChatThreadClientState;
     };
+    latestErrors: ChatErrors;
 };
 
+// @public
+export class ChatError extends Error {
+    constructor(target: ChatErrorTargets, inner: Error);
+    inner: Error;
+    target: ChatErrorTargets;
+}
+
+// @public
+export type ChatErrors = {
+    [target in ChatErrorTargets]: Error;
```

Comment:
Just to clarify this is a more constrained form of Record<string, Error> that ensures the string is an actual method from ChatClient or ChatThreadClient? If yes, that's pretty cool.

Reply:
Yep, that's all it is.

And you get auto-complete in vscode when you look for errors in the state :)